llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -827,6 +827,16 @@ static unsigned getOpcodeForRecipe(VPRecipeBase &R) {
 
 /// Try to simplify recipe \p R.
 static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
+  // Try to remove redundant blend recipes.
+  if (auto *Blend = dyn_cast<VPBlendRecipe>(&R)) {
```

Comment:
nit: may be worth defining and reusing `IncomingValue0 = Blend->getIncomingValue(0);`

Reply:
Done, thanks!